spark/src/main/java/org/apache/spark/sql/comet/CometScalarSubquery.java

```diff
@@ -47,10 +47,12 @@ public static synchronized void setSubquery(long planId, ScalarSubquery subquery
   }
 
   public static synchronized void removeSubquery(long planId, ScalarSubquery subquery) {
-    subqueryMap.get(planId).remove(subquery.exprId().id());
+    if (subqueryMap.containsKey(planId)) {
+      subqueryMap.get(planId).remove(subquery.exprId().id());
```

Comment:
I hit NPE here while working sort merge join PR. I think it is possible that the query plan has duplicate subqueries (e.g., reused subquery), so if it may try to do removal of subquery after the map was clean up.

Reply:
It maybe hard to create a test query without join to reproduce it. So I leave a test out here. But this change looks straightforward.